### PR TITLE
[MNT] Improve coverage

### LINKED ISSUE #93: Improve test coverage
Looks like the main parts to improve are `annotations` submodule. We should also ignore the `commands` submodule probably. I think getting those two fixed should bring our test coverage to 90+

Screen Shot 2022-08-05 at 11 01 22 AM

### PR DESCRIPTION
**Pull request by mscheltienne**
First step for #93.
It's just a start for now ;)

### CODE REVIEW

**Comment by mscheltienne** [comment]
```
Name                                       Stmts   Miss Branch BrPart     Cover   Missing
-----------------------------------------------------------------------------------------
mne_icalabel\iclabel\config.py                 2      0      2      0   100.00%
mne_icalabel\iclabel\features.py             205      0     50      1    99.61%   317->320
mne_icalabel\iclabel\label_components.py      23      0     10      0   100.00%
mne_icalabel\iclabel\network.py               97      0      8      0   100.00%
mne_icalabel\iclabel\utils.py                 96      7     16      3    91.07%   208-213, 239, 247
-----------------------------------------------------------------------------------------
TOTAL                                        423      7     86      4    97.84%
```

For `features.py`, I'm not sure what the branch coverage is complaining about `317 -> 320`. https://github.com/mne-tools/mne-icalabel/blob/07beb68637dd9c0e962492f87a8f8fe8d8857023/mne_icalabel/iclabel/tests/test_features.py#L75-L119 is running though both https://github.com/mne-tools/mne-icalabel/blob/07beb68637dd9c0e962492f87a8f8fe8d8857023/mne_icalabel/iclabel/features.py#L312-L319

File: mne_icalabel/datasets/icalabel/icalabel.py [hunk 1]
```diff
@@ -57,13 +62,4 @@ def data_path(path=None, force_update=False, update_path=True, download=True, ve
...
     )
-    dpath = str(dpath)
-
-    # Do some wrangling to deal with nested directories
-    bad_name = os.path.join(dpath, "mne-testing-icalabel-data-main")
-    if os.path.isdir(bad_name):
-        os.rename(bad_name, dpath + ".true")
-        shutil.rmtree(dpath)
-        os.rename(dpath + ".true", dpath)
-
```
Comment (mscheltienne) [inline mne_icalabel/datasets/icalabel/icalabel.py:68]: @adam2392 Why is this required? The `if` statement is not tested, and locally I can't figure out a case in which I get into it.

Comment (mscheltienne) [reply mne_icalabel/datasets/icalabel/icalabel.py:68]: Nevermind.

**Comment by mscheltienne** [comment]
Posted on the [Codecov discourse](https://community.codecov.com/t/codecov-website-can-not-show-line-by-line-coverage/3809) about the line-by-line display on the website that is failing. I've been looking for a while.. and I can't figure out how to configure it properly.

Screenshot 2022-08-08 at 13 44 59

File: mne_icalabel/iclabel/label_components.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def iclabel_label_components(inst: Union[BaseRaw, BaseEpochs], ica: ICA, inplace
 
         # add labels to the ICA instance
         for idx, (_, mne_label) in enumerate(ICLABEL_LABELS_TO_MNE.items()):
-            auto_labels = np.argwhere(argmax_labels == idx)
+            auto_labels = list(np.argwhere(argmax_labels == idx).flatten())
```
Comment (mscheltienne) [inline mne_icalabel/iclabel/label_components.py:60]: @adam2392 Note that I changed the format from an array to a list here. Let me know if that causes an issue in your opinion.

**Comment by mscheltienne** [comment]
Currently: 

```
---------- coverage: platform win32, python 3.8.10-final-0 -----------                                                                
Name                                         Stmts   Miss Branch BrPart     Cover   Missing                                           
-------------------------------------------------------------------------------------------                                           
mne_icalabel\annotation\bids.py                 61      9     30     10    79.12%   43, 54->52, 89, 121, 127, 136, 145, 150, 152, 155 
mne_icalabel\config.py                           3      0      0      0   100.00%                                                     
mne_icalabel\datasets\icalabel\icalabel.py      19      0      2      0   100.00%                                                     
mne_icalabel\features\topomap.py                48      0      8      0   100.00%                                                     
mne_icalabel\gui\_label_components.py          144     13     36      5    86.67%   60, 68, 77-79, 243-244, 250, 256-258, 274-275     
mne_icalabel\iclabel\config.py                   2      0      2      0   100.00%                                                     
mne_icalabel\iclabel\features.py               205      0     50      1    99.61%   317->320                                          
mne_icalabel\iclabel\label_components.py        23      0     10      0   100.00%                                                     
mne_icalabel\iclabel\network.py                 97      0      8      0   100.00%                                                     
mne_icalabel\iclabel\utils.py                   96      7     16      3    91.07%   208-213, 239, 247                                 
mne_icalabel\label_components.py                22      0      4      0   100.00%                                                     
mne_icalabel\utils\_checks.py                   30      8      8      1    65.79%   45-48, 52-53, 55-56                               
mne_icalabel\utils\_docs.py                     55      0     22      0   100.00%                                                     
-------------------------------------------------------------------------------------------                                           
TOTAL                                          805     37    196     20    93.31%                                                     
```

File: mne_icalabel/features/topomap.py [hunk 3]
```diff
@@ -17,69 +19,87 @@ def get_topomaps(
     ica: ICA,
     picks=None,
     res: int = 64,
-    outlines: Optional[str] = "head",
     image_interp: str = _INTERPOLATION_DEFAULT,  # 'cubic'
     border: Union[float, str] = _BORDER_DEFAULT,  # 'mean'
-    extrapolate: str = _EXTRAPOLATE_DEFAULT,  # 'head'
-):
+    extrapolate: str = _EXTRAPOLATE_DEFAULT,  # 'auto' -> 'head' for EEG, 'local' for MEG
+) -> Dict[str, NDArray[float]]:
```
Comment (adam2392) [inline mne_icalabel/features/topomap.py:25]: Not sure why but this breaks numpydoc validation.

Comment (mscheltienne) [reply mne_icalabel/features/topomap.py:25]: Yes.. I have no idea either. Looking into it.

**Comment by mscheltienne** [comment]
Also, I think the config of `codecov` is a bit off, but I don't know yet how to fix it. It will probably not report the same numbers as the one I got locally.

File: mne_icalabel/features/topomap.py (hunk 3 above)
Comment (mscheltienne) [reply mne_icalabel/features/topomap.py:25]: Now it's showing also:

```
WARNING: [numpydoc] Validation warnings while processing docstring for 'mne_icalabel.features.get_topomaps':
  RT03: Return value has no description

WARNING: [numpydoc] Validation warnings while processing docstring for 'mne_icalabel.features.get_topomaps':
  RT03: Return value has no description
```

Which is (probably?) related.

Comment (mscheltienne) [reply mne_icalabel/features/topomap.py:25]: Solved, It was coming from `sphinx_autodoc_typehints` which was not playing nicely with `numpydoc`.
I simply removed it, type-hints on the rendered website, even only in the signature, can become very messy.

**Comment by mscheltienne** [comment]
@adam2392 Good to merge on my end if CIs are green. For the `annotation` module, you can work on that one in another PR?

File: doc/whats_new.rst [hunk 4]
```diff
@@ -24,7 +24,9 @@ Version 0.3 (Unreleased)
 Enhancements
 ~~~~~~~~~~~~
 
-- Adding a GUI to facilitate the labeling of ICA components, by `Adam Li`_ and `Mathieu Scheltienne`_ (:gh:`66`)
+- Adding a GUI to facilitate the labeling of ICA components by `Adam Li`_ and `Mathieu Scheltienne`_ (:gh:`66`)
+
```
Comment (adam2392) [inline doc/whats_new.rst:28]:
```suggestion
```

**Review by adam2392 — APPROVED** [review]

**Pull request merged**